tests/core/management/schedulers/test_master_scheduler.py

```diff
@@ -181,3 +187,14 @@ async def test_schedule_interrupt_succeeds_with_queued_wakeup(
         "me first": master_scheduler.ticker.time,
         "bar": 42,
     }
+
+
+@pytest.mark.asyncio
+async def test_master_scheduler_handle_exception_message(
```

Comment:
This duplicates the testing on the `BaseScheduler`. Concrete class tests should test behaviour, not implementation. Would it be possible to test that the master scheduler produces the correct side effects (i.e. kill commands, log messages & program exit) when it receives an error message

Reply:
There's an argument for just getting rid of the testing on `BaseScheduler` and relying on this to also test the base functionality. What do you think?

Reply:
Yeah, that's not a bad way to go, though I would put that off until we inevitably switch to a composition based model